packages/std/src/math/decimal.rs

```diff
@@ -406,15 +404,20 @@ impl FromStr for Decimal {
             let fractional = fractional_part
                 .parse::<Uint128>()
                 .map_err(|_| StdError::generic_err("Error parsing fractional"))?;
-            let exp =
-                (Self::DECIMAL_PLACES.checked_sub(fractional_part.len())).ok_or_else(|| {
-                    StdError::generic_err(format!(
-                        "Cannot parse more than {} fractional digits",
-                        Self::DECIMAL_PLACES
-                    ))
-                })?;
+            let exp = (Self::DECIMAL_PLACES.checked_sub(
+                fractional_part
+                    .len()
+                    .try_into()
+                    .map_err(|_| StdError::generic_err("fractional too long"))?,
```

Comment:
Let's panic here or even do an unsafe cast from `usize` to `u32`. In wasm32, `.len()` can never exceed the u32 range (4GB of string data).